homework/index.js

```diff
@@ -30,17 +24,156 @@
+        text: 'Contributors',
+        class: 'contributors-title',
+      });
+      if (!contributors.length) {
+        contributorsTitle.textContent = 'No Contributor So Far';
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      const contributorsList = createAndAppend('ul', rightSide, {
+        class: 'contributors-list',
+      });
+      contributors.forEach(contributor => {
+        const listItem = createAndAppend('li', contributorsList, {
+          class: 'contributor',
+          tabindex: 0,
+          'aria-label': contributor.login,
+        });
+        createAndAppend('img', listItem, {
+          class: 'contributor-avatar',
+          src: contributor.avatar_url,
+        });
+        createAndAppend('span', listItem, {
+          class: 'contributor-name',
+          text: contributor.login,
+        });
+        createAndAppend('span', listItem, {
+          class: 'contribution-count',
+          text: contributor.contributions,
+        });
+        listItem.addEventListener('click', () => {
+          // Go to new page on click
+          window.open(contributor.html_url, '_blank');
+        });
+        listItem.addEventListener('keyup', event => {
+          // Open new page with contributor url when pressed Enter on focused contributor
+          if (event.key === 'Enter') window.open(contributor.html_url, '_blank');
+        });
+      });
+    } catch (error) {
+      renderError(error);
+    }
+  }
+
+  function initializePage(repositories) {
+    const select = document.getElementById('select');
+    const wrapper = document.getElementById('wrapper');
+    repositories.sort((a, b) => a.name.localeCompare(b.name));
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        value: index,
+        text: repository.name,
+      });
+    });
+    createAndAppend('section', wrapper, {
+      class: 'left-side',
+      id: 'left-side',
+    });
+    renderRepoInfo(repositories[select.value]);
+    createAndAppend('section', wrapper, {
+      class: 'right-side',
+      id: 'right-side',
+    });
+    select.addEventListener('change', () => {
+      renderRepoInfo(repositories[select.value]);
+      fetchAndRender(repositories[select.value]);
+    });
+    return repositories[select.value];
+  }
+
+  async function renderPage(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('h1', header, {
+      class: 'header-title',
+      text: 'Hack Your Future Repositories',
+    });
+    createAndAppend('main', root, {
+      id: 'wrapper',
     });
+    createAndAppend('select', header, {
+      id: 'select',
+      'aria-label': 'HYF Repositories',
+    });
+
+    try {
+      const repositories = await fetchJSON(url);
+      const repository = initializePage(repositories);
```

Comment:
It is not immediately obvious what `initializePage` returns or should return. The function name gives no clue. I had to inspect the code to discover that it effectively returns the first repo from the list. Therefore, I would prefer this explicit code:

```js
try {
  const repositories = await fetchJSON(url);
  initializePage(repositories);
  fetchAndRender(repositories[0]);
} catch (error) {
  renderError(error);
}
```

Reply:
Changed following your advice